pkg/controller/servicebinding/binder.go

```diff
@@ -161,28 +170,24 @@ func (b *binder) updateVolumes(volumes []interface{}) ([]interface{}, error) {
 	name := b.sbr.GetName()
 	log := b.logger
 
-	// FIXME(isuttonl): update should not bail out here since b.volumeKeys might change
 	log.Debug("Checking if binding volume is already defined...")
 	for _, v := range volumes {
-		volume := v.(corev1.Volume)
-		if name == volume.Name {
+		volume, ok := v.(map[string]interface{})
+		if !ok {
+			return nil, &typeAssertion{}
```

Comment:
do we really need here a new type? We had previously error. I asked just to improve the error message, so if it happens that we know what is going on and where.